BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,437 @@
+/**********************************************************************
+Copyright (c) 2016 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+********************************************************************/
+
+#include "render.h"
+#include "scene_io.h"
+#include "material_io.h"
+#include "SceneGraph/light.h"
+#include "Output/clwoutput.h"
+#include "BaikalIO/image_io.h"
+
+#include <fstream>
+#include "XML/tinyxml2.h"
+
+using namespace Baikal;
+
+static std::uint32_t constexpr kNumIterations = 4096;
+
+struct OutputDesc
+{
+    Renderer::OutputType type;
+    std::string name;
+    // extansion of the file to save
+    std::string file_ext;
+    int bpp;
+};
+
+// if you need to add new output for saving to disk
+// just put its description in thic collection
+std::vector<OutputDesc> outputs_collection = { { Renderer::OutputType::kColor, "color", "png", 16 },
```

Comment:
Is PNG applicable for such HDR images as color and depth ?

Reply:
Not actual